service/templates/cloudformation/nat_gateway.yaml

```diff
@@ -0,0 +1,21 @@
+{{define "nat_gateway"}}
+  NATGateway:
+    Type: AWS::EC2::NatGateway
+    Properties:
+      AllocationId:
+        Fn::GetAtt:
+        - NATEIP
+        - AllocationId
+      SubnetId: {{.PublicSubnetID}}
```

Comment:
Can we add a Name tag here with the cluster ID? 

We do that in the old AWS resources. We get Cloud Formation tags added automatically with the stack name that has the cluster ID. But the name tag is useful in the AWS console.

Reply:
Sure, done. I've taken the opportunity to promote `ClusterID` to the top level adapter (it was used in ASG too). Also now we don't try to delete NAT gw if on CloudFormation.

Reply:
Thanks, and yes I like that shared fields are moved to the top level adapter.